meta_tests/conftest_test.py

```diff
@@ -45,29 +35,89 @@ def setup_method(self, method):
+        dtest_config.skip_resource_intensive_tests = False
+        dtest_config.use_vnodes = False
+        dtest_config.use_off_heap_memtables = False
+
+        # Configurations making tests without mark to be skipped
+        dtest_config.only_resource_intensive_tests = True
+        assert is_skippable(self.regular_test, dtest_config, True)
+        assert is_skippable(self.regular_test, dtest_config, False)
+        dtest_config.only_resource_intensive_tests = False
+
+        dtest_config.execute_upgrade_tests_only = True
+        assert is_skippable(self.regular_test, dtest_config, True)
+        dtest_config.execute_upgrade_tests_only = False
 
     def test_upgrade_test(self):
-        assert not _is_skippable(item=self.upgrade_test)
-        assert _is_skippable(item=self.upgrade_test, include_upgrade_tests=False)
+        dtest_config = DTestConfigMock()
+        assert is_skippable(self.upgrade_test, dtest_config, True)
+
+        dtest_config.execute_upgrade_tests = True
+        assert not is_skippable(self.upgrade_test, dtest_config, True)
+        dtest_config.execute_upgrade_tests = False
+
+        dtest_config.execute_upgrade_tests_only = True
+        assert not is_skippable(self.upgrade_test, dtest_config, True)
+        dtest_config.execute_upgrade_tests_only = False
 
     def test_resource_intensive_test(self):
-        assert not _is_skippable(item=self.resource_intensive_test)
-        assert _is_skippable(item=self.resource_intensive_test, include_resource_intensive_tests=False)
+        dtest_config = DTestConfigMock()
+        assert not is_skippable(self.resource_intensive_test, dtest_config, True)
+        assert is_skippable(self.resource_intensive_test, dtest_config, False)
+
+        dtest_config.force_execution_of_resource_intensive_tests = True
+        assert not is_skippable(self.resource_intensive_test, dtest_config, True)
+        assert not is_skippable(self.resource_intensive_test, dtest_config, False)
+        dtest_config.force_execution_of_resource_intensive_tests = False
+
+        dtest_config.only_resource_intensive_tests = True
+        assert not is_skippable(self.resource_intensive_test, dtest_config, True)
+        assert not is_skippable(self.resource_intensive_test, dtest_config, False)
+        dtest_config.only_resource_intensive_tests = False
+
+        dtest_config.skip_resource_intensive_tests = True
+        assert is_skippable(self.resource_intensive_test, dtest_config, True)
+        dtest_config.skip_resource_intensive_tests = False
 
     def test_vnodes_test(self):
-        assert not _is_skippable(item=self.vnodes_test)
-        assert _is_skippable(item=self.vnodes_test, include_vnodes_tests=False)
+        dtest_config = DTestConfigMock()
+        assert is_skippable(self.vnodes_test, dtest_config, True)
+
+        dtest_config.use_vnodes = True
+        assert not is_skippable(self.vnodes_test, dtest_config, True)
+        dtest_config.use_vnodes = False
```

Comment:
this is very bad practice... each test should prepare new mock, because..
... if something goes wrong with this one you will not get to the line:
```
dtest_config.use_vnodes = False
```
and failure of one test case will leave whole suite in a unknown state.

Instead you can use some test fixttures probably to return new mock for every case.